### PR TITLE
Fix article save2copy not unpublished

### LINKED ISSUE #37351: [J 4.1.x] Save as copy saves article published
### Steps to reproduce the issue
In the previous Joomla version Save as Copy task saved the article as copy but unpublished
Now the Save as Copy task automatically publishes the article

### Expected result
When do you want to save something as a copy? Probably when you want to take the existing one as a draft for the next one. In this case you would not want to have it published until you are finished with your editing. In Joomla 4 somehow this changed.

### Actual result
When you open an article and click on the Save as Copy Button, the copy is created and published.

### System information (as much as possible)
Joomla 4.1.0

### Additional comments
Reported by Dominik on Facebook, I agree that this is not user friendly

### PR DESCRIPTION
**Pull request by kitepascal**
Pull Request for Issue #37351 .

### Testing Instructions
Save as Copy

### Actual result BEFORE applying this Pull Request
Article is published.

### Expected result AFTER applying this Pull Request
Article shoud be unpublished like in J3 before.

### Documentation Changes Required
None.

Fixes #37351

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on b17c1c5bf819b19520b8cd0f7e4208f8bee6685e<br /><br />Tested successfully in 4.1.1-rc2-dev of 23 March using PHP 8.0.15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37354">issues.joomla.org/tracker/joomla-cms/37354</a>.</sub>

**Comment by Shubhamverma2796** [comment]
I have tested this item :white_check_mark: successfully on b17c1c5bf819b19520b8cd0f7e4208f8bee6685e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37354">issues.joomla.org/tracker/joomla-cms/37354</a>.</sub>

File: administrator/components/com_content/src/Model/ArticleModel.php [hunk 1]
```diff
@@ -784,6 +784,7 @@ public function save($data)
 		if ($input->get('task') == 'save2copy')
 		{
 			$origTable = $this->getTable();
+			$data['state'] = 0;
```
Comment (richard67) [inline administrator/components/com_content/src/Model/ArticleModel.php]:
```suggestion
			$data['state'] = 0;
```
Please remove the space here at the end of the line to fix the code style test failing in drone (our tool for automated checks).

File: administrator/components/com_content/src/Model/ArticleModel.php [hunk 2]
```diff
@@ -784,6 +784,7 @@ public function save($data)
 		if ($input->get('task') == 'save2copy')
```
Comment (richard67) [inline administrator/components/com_content/src/Model/ArticleModel.php]:
```suggestion
		if ($input->get('task') === 'save2copy')
```
Use type safe comparison if possible,

**Comment by brianteeman** [comment]
I am a little unsure that this is the correct fix. It would be useful to find out when/where the j3 behaviour was changed. Was there a reason for it, was anything else changed at the same time. There may be more needed than just this one line.

**Comment by richard67** [comment]
> I am a little unsure that this is the correct fix. It would be useful to find out when/where the j3 behaviour was changed. Was there a reason for it, was anything else changed at the same time. There may be more needed than just this one line.

Yes, I have the same feeling, but I'm at work so can't investigate. I hope someone else can step in if necessary.

**Comment by brianteeman** [comment]
Well there is no rush to merge it ;)

Thinking aloud but it might be something to do with the creation of multilingual associated articles

**Comment by kitepascal** [comment]
> it might be something to do with the creation of multilingual associated articles

Created multilingual associated articles are still published initially, just tested that.
Also if you create an associated translation of an unpublished article.

**Comment by laoneo** [comment]
Perhaps @bembelimen can shed some light in here as it can relate to the workflow changes.

**Comment by richard67** [comment]
> Perhaps @bembelimen can shed some light in here as it can relate to the workflow changes.

I think that could be the point. When workflow is enabled, it should use the default stage, which might be different. Maybe we should look up in the code how it's done for batch copying?

**Comment by bembelimen** [comment]
Workflow didn't change the behaviour... I wanted to do so, but there was too much restistant and for me it was not worth the time to fight for it...so I think copy should always save unpublished, but that is just my opinion.

**Comment by brianteeman** [comment]
Well something changed it - weird

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Comment by Quy** [comment]
Closing in favor of #38810 as it has more in-depth discussion. Thank you for your contribution.

**Pull request closed without merging**